src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/service/ListingService.java

```diff
@@ -22,19 +26,30 @@
     private final Logger log = LoggerFactory.getLogger(ListingService.class);
 
     private final ListingRepository listingRepository;
+    private final LocationRepository locationRepository;
+    private final UserRepository userRepository;
 
-    @Autowired public ListingService(@Qualifier("listingRepository")ListingRepository listingRepository) {
+    @Autowired public ListingService(@Qualifier("listingRepository") ListingRepository listingRepository, LocationRepository locationRepository, UserRepository userRepository) {
         this.listingRepository = listingRepository;
+        this.locationRepository = locationRepository;
+        this.userRepository = userRepository;
     }
 
     public List<Listing> getListings() {
         return this.listingRepository.findAll();
     }
 
-    public @NotNull Listing createListing(Listing newListing) {
+    public @NotNull Listing createListing(Listing newListing, Location address, UUID publisherID) {
```

Comment:
shouldn't we rather just use an argument of type Listing and you extract the address and publisher from there to process it? because now you would be able to set the address in newListing and another address in Location address